src/exo1 - Basic types/exo1.ts

```diff
@@ -54,11 +69,11 @@ export const safeDivide: (a: number, b: number) => Option<number> =
 // Here is a simple error type to help you:
 export type DivisionByZeroError = 'Error: Division by zero';
 export const DivisionByZero = 'Error: Division by zero' as const;
-
-export const safeDivideWithError: (
+ 
+export const safeDivideWithError = (
   a: number,
-  b: number,
-) => Either<DivisionByZeroError, number> = unimplemented;
+  b: number) =>  b === 0 ? either.left(DivisionByZero) :
+    either.right(divide(a, b));
```

Comment:
Usually the pattern `condition ? either.left : either.right` is a good indicator of a place where you could use `either.fromPredicate`, so to keep the logic you wrote here, you could write:
```ts
export const safeDivideWithError = (a: number, b: number) =>
  pipe(
    b,
    either.fromPredicate(
      n => n !== 0,
      () => DivisionByZero,
    ),
    either.map(b => a / b),
  );
```